src/fsharp/Optimizer.fs

```diff
@@ -2290,14 +2302,15 @@ and OptimizeLinearExpr cenv env expr contf =
               Info = evalue' } ))
 
     | LinearMatchExpr (spMatch, exprm, dtree, tg1, e2, spTarget2, m, ty) ->
-         let dtree, dinfo = OptimizeDecisionTree cenv env m dtree
+         let dtree', dinfo = OptimizeDecisionTree cenv env m dtree
```

Comment:
Just recording this as an idle thought, gosh it'd be nice if we had the time to build a proper expression evaluator so that `foo'` identifiers and shadowed identifiers could get picked up in debug tools properly

Reply:
Yes.  We should also just remove all use of `foo'` identifiers from the compiler